notebooks/using-mlflow/logging-models/logging_model_with_mlflow.ipynb

```diff
@@ -0,0 +1,600 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f5a4d2ef",
+   "metadata": {},
+   "source": [
+    "### Prerequisites to run this notebook"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fccfe0c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Ensure you have the dependencies for this notebook\n",
+    "%pip install -r logging_model_with_mlflow.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c384f488",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import mlflow\n",
+    "\n",
+    "mlflow.set_experiment(\"heart-disease-classifier\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1fae6e41",
+   "metadata": {},
+   "source": [
+    "Reading the data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e2c361e6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "file_url = \"http://storage.googleapis.com/download.tensorflow.org/data/heart.csv\"\n",
+    "df = pd.read_csv(file_url)\n",
+    "df[\"thal\"] = df[\"thal\"].astype(\"category\").cat.codes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9df947eb",
+   "metadata": {},
+   "source": [
+    "Let's do some basic train-test split:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e94b3d25",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "X_train, X_test, y_train, y_test = train_test_split(\n",
+    "    df.drop(\"target\", axis=1), df[\"target\"], test_size=0.3\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e288761d",
+   "metadata": {},
+   "source": [
+    "### Logging models using `autolog()`\n",
+    "\n",
+    "One of the simplest ways to start using this approach is by using MLflow autolog functionality. Autolog allows MLflow to instruct the framework associated to with the framework you are using to log all the metrics, parameters, artifacts and models that the framework considers relevant. By default, most models will be log if autolog is enabled. Some flavors may decide not to do that in specific situations. For instance, the flavor PySpark won't log models if they exceed a certain size. \n",
+    "\n",
+    "You can turn on autologging by using the autolog method for the framework you are using. For instance, for XGBoost models,\n",
+    "\n",
+    "```python\n",
+    "mlflow.xgboost.autolog()\n",
+    "```\n",
+    "\n",
+    "Our training would look like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f215ea43",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import mlflow\n",
+    "from xgboost import XGBClassifier\n",
```

Comment:
I just verified what happened. The problem is that the Azure ML kernel has tons of libraries installed and the installation actually fails because versions collide. I will remove fixing the versions so we can be more "grateful" with versions, but this may make the samples fail unexpectedly because of the versions the user may have.